src/types.ts

```diff
@@ -1,6 +1,9 @@
-export type Validator<T = any, E = any> = (subject: T | undefined, context?: object) => E | undefined
-export type SubjectType<V extends Validator> = Parameters<V>[0]
-
-export type PrimitiveErrors = Array<string>
-export type ObjectTopLevelError<E> = { _self: E }
+// deno-lint-ignore ban-types no-explicit-any
+export type Validator<T = any, E = any, C extends {} = {}> = (
```

Comment:
I find `T` (and `E` as well tbh) to be a lacking name for a Type.
I would prefer `Subject` or `Error`. (`Context`) as well

Reply:
I agree, old code being old. Will change it.